### PR TITLE
feat: Support optional headers for OAuth request

### PR DESCRIPTION
**Pull request by s7clarke10**
Adding support for adding a header to an OAuth request for tokens. In a few instances an API has a requirement to have headers added to the request for an OAuth Token. An example of the type of header is a subscription key.

This extends the current OAuth Authenticator to accept a new **optional** parameter `oauth_headers`.

The PR also explicitly converts the returned `expires_in` value to int rather than assuming it is an integer. I had an example where it was returned as an string from this API.

----
:books: Documentation preview :books:: https://meltano-sdk--1815.org.readthedocs.build/en/1815/

### CODE REVIEW

**Review by edgarrmondragon — COMMENTED** [review]
Thanks @s7clarke10! A few suggestions to simplify things a bit, let me know what you think :)

File: singer_sdk/authenticators.py [hunk 1]
```diff
@@ -512,7 +518,7 @@ def update_access_token(self) -> None:
 
         token_json = token_response.json()
         self.access_token = token_json["access_token"]
-        self.expires_in = token_json.get("expires_in", self._default_expiration)
+        self.expires_in = int(token_json.get("expires_in", self._default_expiration))
```
[the remark this answers is not in the document]
Comment (s7clarke10) [reply singer_sdk/authenticators.py]: Great suggestions, I was looking at similar code based on the failed test but ran out of time of Friday - really appreciate you feedback. I have adopted these changes.

File: singer_sdk/authenticators.py [hunk 2]
```diff
@@ -377,6 +378,10 @@ def __init__(
         self.refresh_token: str | None = None
         self.last_refreshed: datetime | None = None
         self.expires_in: int | None = None
+        if self._auth_headers is None:
+            self._auth_headers = {}
+        if oauth_headers:
+            self._auth_headers.update(oauth_headers)
```
[the remark this answers is not in the document]
Comment (s7clarke10) [reply singer_sdk/authenticators.py]: Yes, you are absolutely right - we should be using separate oauth_headers.

I have made a slight tweak to align with the parameter name. I have also provided an appropriate args comment for the OAuth function oauth_headers parameter.

```python
        self._oauth_headers = oauth_headers or {}
```

File: singer_sdk/authenticators.py [hunk 3]
```diff
@@ -499,6 +504,7 @@ def update_access_token(self) -> None:
         auth_request_payload = self.oauth_request_payload
         token_response = requests.post(
             self.auth_endpoint,
+            headers=self._auth_headers,
```
[the remark this answers is not in the document]
Comment (s7clarke10) [reply singer_sdk/authenticators.py]: Agreed, we need to adopt this new name.

**Comment by s7clarke10** [comment]
This PR has been tested on my enhancements to tap-rest-api-msdk. This is the branch that I have used for my testing https://github.com/s7clarke10/tap-rest-api-msdk/commits/feature/sdk_supported_oauth_headers .

Thanks for your support for this change and great suggestions to resolve the failed tests. I believe all the comments can be resolved now.

**Comment by s7clarke10** [comment]
Fixes https://github.com/meltano/sdk/issues/1822

**Review by edgarrmondragon — APPROVED** [review]
Thanks @s7clarke10!

**Pull request merged**